From 04fce6938591a422a49fc6687b8788566046e080 Mon Sep 17 00:00:00 2001
From: Gu, Wangyi <wangyi.gu@intel.com>
Date: Tue, 15 Jan 2013 10:01:22 +0800
Subject: Movie studio: fix a memory leak issue in the AOSP code

BZ: 80150

It's an android defect, connect()/disconnect() of OMXClient need be called in a pair,
if miss calling disconnect() while connect() is called, a memory leak will happen.

Category:aosp improvement
Domain: Video.Media-video editor
Origin: internal
Upstream-Candidate: No

Change-Id: I91f6c6af9b6567b4af73fe4cfbed118edcaaccbd
Orig-Change-Id: I9acc19242faa5c981bb77caba52170e0f96dbf96
Signed-off-by: Gu, Wangyi <wangyi.gu@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
Signed-off-by: chuzhoux <chux.zhou@intel.com>
---
 .../src/VideoEditorVideoDecoder.cpp                |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/libvideoeditor/vss/stagefrightshells/src/VideoEditorVideoDecoder.cpp b/libvideoeditor/vss/stagefrightshells/src/VideoEditorVideoDecoder.cpp
index de91731..1845480 100755
--- a/libvideoeditor/vss/stagefrightshells/src/VideoEditorVideoDecoder.cpp
+++ b/libvideoeditor/vss/stagefrightshells/src/VideoEditorVideoDecoder.cpp
@@ -779,6 +779,7 @@ M4OSA_ERR queryVideoDecoderCapabilities
     logSupportDecodersAndCapabilities(pDecoders);
     *decoders = pDecoders;
 cleanUp:
+    client.disconnect();
     return err;
 }
 /********************
-- 
1.7.4.1

